api/v1alpha1/hostedcluster_types.go

```diff
@@ -42,10 +42,65 @@ type HostedClusterSpec struct {
 
 	SSHKey corev1.LocalObjectReference `json:"sshKey"`
 
+	// ProviderCreds is a reference to a secret containing cloud account info
 	ProviderCreds corev1.LocalObjectReference `json:"providerCreds"`
 
+	// Networking contains network-specific settings for this cluster
+	Networking ClusterNetworking `json:"networking"`
+
+	Platform PlatformSpec `json:"platform"`
+
+	// InfraID is used to identify the cluster in cloud platforms
+	InfraID string `json:"infraID,omitempty"`
```

Comment:
Just for the record I'm not a fan of the UX exposing the infraID in the API. I think we could infer it from the VPC ID and let that be the only contract.
Discussed with @csrwng to go as in this PR for now.

Reply:
Should definitely do a follow up. Right now specifying the infra ID allows me to avoid doing an AWS call from the hypershift operator to discover it. I can pass it directly to the control plane where it's needed for the cloud config.